task-sdk/src/airflow/sdk/exceptions.py

```diff
@@ -68,3 +69,22 @@ def __str__(self) -> str:
         for arg in self.args[1:]:
             typename = f"{typename}[{type(arg).__qualname__}]"
         return f"unmappable return type {typename!r}"
+
+
+class FailFastDagInvalidTriggerRule(AirflowException):
+    """Raise when a dag has 'fail_fast' enabled yet has a non-default trigger rule."""
+
+    _allowed_rules = (TriggerRule.ALL_SUCCESS, TriggerRule.ALL_DONE_SETUP_SUCCESS)
+
+    @classmethod
+    def check(cls, *, fail_fast: bool, trigger_rule: TriggerRule):
+        """
+        Check that fail_fast dag tasks have allowable trigger rules.
+
+        :meta private:
+        """
+        if fail_fast and trigger_rule not in cls._allowed_rules:
+            raise cls()
+
+    def __str__(self) -> str:
+        return f"A 'fail_fast' dag can only have {TriggerRule.ALL_SUCCESS} trigger rule"
```

Comment:
As I understand this, its only raised and not caught in the sdk. If that's the case, you should not import this in the core (even in tests), move those tests to sdk.